src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlStoreClient.cs

```diff
@@ -119,8 +119,8 @@ internal async Task<IReadOnlyList<ResourceWrapper>> GetResourcesByTransactionIdA
         {
             using var cmd = new SqlCommand() { CommandText = "dbo.GetResourcesByTransactionId", CommandType = CommandType.StoredProcedure, CommandTimeout = 600 };
             cmd.Parameters.AddWithValue("@TransactionId", transactionId);
-            //// ignore nulls returned for invisible resources
-            return (await cmd.ExecuteReaderAsync(_sqlRetryService, (reader) => { return ReadResourceWrapper(reader, true, decompress, getResourceTypeName); }, _logger, cancellationToken)).Where(_ => _ != null).ToList();
+            //// ignore invisible resources
+            return (await cmd.ExecuteReaderAsync(_sqlRetryService, (reader) => { return ReadResourceWrapper(reader, true, decompress, getResourceTypeName); }, _logger, cancellationToken)).Where(_ => _.RawResource.Data != _invisibleResource).ToList();
```

Comment:
NIT: can __.RawResource_ be null under this context?

Reply:
No. This is changed in the reader, such as " " is plugged in for data for invisible records.